include/rocksdb/write_batch_base.h

```diff
@@ -42,6 +42,13 @@ class WriteBatchBase {
                      const SliceParts& value);
   virtual Status Put(const SliceParts& key, const SliceParts& value);
 
+  // Store the mapping "key->value" in the database with the specified write
+  // time in the default column family.
```

Comment:
For generality, I think we should expect a ColumnFamilyHandle.

I would add some "best effort" language, as in "The DB makes a reasonable best effort to treat the data as having the given write time for the purposes of data placement and preservation, but doesn't currently make any guarantees."

And we might need to think about the semantics of Put(a, b) then later TimedPut(a, c, some time before first Put). (a) Is c guaranteed to remain the latest value (until next overwrite)?
(b) Is it generally undefined behavior to TimedPut on top of an existing key?
(c) Is it just undefined which takes precendence and that can change over time (due to compactions)?

I think (c) gives us the plenty of flexibility while promising the user that they get a valid answer even if there could be more than one "valid" answer. (In practice, I believe this resembles SingleDelete after more than one Put on a key. We essentially use (b) for SingleDelete but probably could promise more like (c).)

Reply:
Thank you for the detailed suggestions! I have updated the documentation for this API. The flexibility of (c) indeed makes it more desirable.